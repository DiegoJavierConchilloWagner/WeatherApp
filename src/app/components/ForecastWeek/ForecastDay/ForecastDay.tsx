import { FC } from 'react';
import dayjs from 'dayjs';

import { IWeekMapped, kelvinToCelcius } from '@common/index';
import { WeatherIcon } from '../../WeatherIcon/WeatherIcon';
import { Icon, IconContainer, StyledForecastedDay, TemperaturesContainer } from '../styled';

interface Props {
	day: IWeekMapped;
	isLoading: boolean;
}

export const ForecastDay: FC<Props> = ({ day, isLoading }) => {
	const { id, icon, main, dt } = day;
	const date = dayjs.unix(dt);
	return (
		<StyledForecastedDay>
			<h3>{date.format('dddd')}</h3>
			<IconContainer>
				<Icon>
					<WeatherIcon id={id} icon={icon} size={80} isLoading={isLoading} />
					<h5>{main}</h5>
				</Icon>
				<TemperaturesContainer>
					<h3>{kelvinToCelcius(day.tempMax)}</h3>
					<h4>{kelvinToCelcius(day.tempMin)}</h4>
				</TemperaturesContainer>
			</IconContainer>
		</StyledForecastedDay>
	);
};
