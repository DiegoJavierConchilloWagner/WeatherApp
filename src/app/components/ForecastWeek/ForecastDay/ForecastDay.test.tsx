import renderWithThemeAndProvider from '@common/utils/renderWhitThemeAndProvider';
import { ForecastDay } from './ForecastDay';

const day = {
	dt: 0,
	feelsLike: 0,
	tempMin: 0,
	tempMax: 0,
	description: '',
	id: 0,
	main: '',
	icon: ''
};

describe('Example ForecastSwitch', () => {
	it('should render correcty', () => {
		const wrapper = renderWithThemeAndProvider(<ForecastDay day={day} isLoading={false} />);

		expect(wrapper).toMatchSnapshot();
	});
});
