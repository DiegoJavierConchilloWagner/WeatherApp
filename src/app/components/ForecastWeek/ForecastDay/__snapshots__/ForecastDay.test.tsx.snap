// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example ForecastSwitch should render correcty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-scroll-snap-align: start;
  -moz-scroll-snap-align: start;
  -ms-scroll-snap-align: start;
  scroll-snap-align: start;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  font-family: Poppins;
  text-align: center;
}

.c0 h4 {
  font-weight: 400;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 h5 {
  font-weight: 500;
  text-align: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width:768px) {
  .c0:not(:last-child) {
    border-bottom: 1px solid white;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <h3>
          Wednesday
        </h3>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <img
              alt="weather"
              height="80"
              src="SvgrURL"
              width="80"
            />
            <h5 />
          </div>
          <div
            class="c3"
          >
            <h3>
              -273
            </h3>
            <h4>
              -273
            </h4>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-scroll-snap-align: start;
  -moz-scroll-snap-align: start;
  -ms-scroll-snap-align: start;
  scroll-snap-align: start;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  font-family: Poppins;
  text-align: center;
}

.c0 h4 {
  font-weight: 400;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 h5 {
  font-weight: 500;
  text-align: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width:768px) {
  .c0:not(:last-child) {
    border-bottom: 1px solid white;
  }
}

<div>
    <div
      class="c0"
    >
      <h3>
        Wednesday
      </h3>
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <img
            alt="weather"
            height="80"
            src="SvgrURL"
            width="80"
          />
          <h5 />
        </div>
        <div
          class="c3"
        >
          <h3>
            -273
          </h3>
          <h4>
            -273
          </h4>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
